internal/pkg/deploy/cloudformation/stack/cc_pipeline_integration_test.go

```diff
@@ -0,0 +1,56 @@
+// +build integration
+
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package stack_test
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+	"github.com/aws/copilot-cli/internal/pkg/deploy/cloudformation/stack"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCC_Pipeline_Template ensures that the CloudFormation template generated for a pipeline matches our pre-defined template.
+func TestCC_Pipeline_Template(t *testing.T) {
+	ps := stack.NewPipelineStackConfig(&deploy.CreatePipelineInput{
+		AppName: "phonetool",
+		Name:    "phonetool-pipeline",
+		Source: &deploy.CodeCommitSource{
+			ProviderName:  manifest.CodeCommitProviderName,
+			RepositoryURL: "https://us-west-2.console.aws.amazon.com/codesuite/codecommit/repositories/aws-sample/browse",
+			Branch:        "master",
+		},
+		Stages: []deploy.PipelineStage{
+			{
+				AssociatedEnvironment: &deploy.AssociatedEnvironment{
+					Name:      "test",
+					Region:    "us-west-2",
+					AccountID: "1111",
+				},
+				LocalWorkloads:   []string{"api"},
+				RequiresApproval: false,
+				TestCommands:     []string{`echo "test"`},
+			},
+		},
+		ArtifactBuckets: []deploy.ArtifactBucket{
+			{
+				BucketName: "fancy-bucket",
+				KeyArn:     "arn:aws:kms:us-west-2:1111:key/abcd",
+			},
+		},
+		AdditionalTags: nil,
+	})
+
+	actual, err := ps.Template()
+	require.NoError(t, err, "template should have rendered successfully")
+	expected, err := ioutil.ReadFile(filepath.Join("testdata", "pipeline", "cc_template.yaml"))
```

Comment:
My slight worry with our integ tests that compare directly strings is that just changing the order of a field, or adding a new comment break the tests 🎻 

1. Ideally, we should be able to just compare the content of the YAML files, ignoring the order or comments. Can you try this and seeing if it works? I don't think it will unmarshal but worth a shot:
```go
m1 := make(map[interface{}]interface{}
require.NoError(t, yaml.Unmarshal(actual, m1))

m2 := make(map[interface{}]interface{}
require.NoError(t, yaml.Unmarshal(wanted, m2))

require.Equal(t, m1, m2)
```

2. If that doesn't work can we do something similar to our autoscaling tests in `stack_integration_test.go`, where we compare strings of only relevant chunks. For example over here we could do something like:

```go
wantedStageContent := `
      Stages:
        - Name: Source
          Actions:
            - Name: SourceCodeFor-phonetool
              ActionTypeId:
                Category: Source
                Owner: AWS
                Version: 1
                Provider: CodeCommit
              Configuration:
                RepositoryName: aws-sample
                BranchName: master
              OutputArtifacts:
                - Name: SCCheckoutArtifact
              RunOrder: 1`
wantedPermissions := `
              - codecommit:GetBranch
              - codecommit:GetCommit
              - codecommit:UploadArchive
              - codecommit:GetUploadArchiveStatus
              - codecommit:CancelUploadArchive`

require.Contains(t, actual, wantedStageContent)
require.Contains(t, actual, wantedPermissions)
```

Reply:
Option 1 works!!! I'll also change the other stack package template test.

Reply:
(I did test both by introducing errors into the testdata templates.)